### PR TITLE
Count and display error details

### LINKED ISSUE #111: Count and display items that failed indexing
### What did you do?
Mass index using IndexCommand
### What did you expect to happen?
Index all items
### What actually happened?
Indexed some items
### What versions of PHP, Lumen and this module are you using?
7.1.20, Lumen 5.6 and this module ^3.3
Please include code that reproduces the issue. The best reproductions are self-contained scripts with minimal dependencies.

```php

```
Some items were not indexed due to mapping error. Would be nice to catch and display count at the end.

### PR DESCRIPTION
**Pull request by wojo1206**
(Please read the [guidelines](.github/CONTRIBUTING.md) before creating PRs.)

Fixes #111

Changes proposed in this pull request:
 * Display detailed errors on failed indexed items

### CODE REVIEW

**Comment by Jalle19** [comment]
@hugovk any idea why Style CI didn't make a PR in this case? Is it only triggered for collaborators or something?

**Comment by hugovk** [comment]
@Jalle19 I reckon they only send PRs for branches in the enabled repo, in this case `digiaonline/lumen-elasticsearch`. 

They don't know anything about (non-enabled) forks, unless of course a PR is sent and they can analyse it here, but don't send PRs to forks.

**Comment by Jalle19** [comment]
Ah of course

File: src/Console/IndexCommand.php [hunk 1]
```diff
@@ -123,4 +132,39 @@ protected function getCount()
     {
         return count($this->getData());
     }
+
+    /**
+     * @param array $responses
+     * @return array
+     */
+    protected function getErrors($responses)
```
Comment (Jalle19) [inline src/Console/IndexCommand.php]: I would move all this to a separate class, maybe `BulkResponseAggregator`, that way you can add some unit tests for it.

Comment (Jalle19) [inline src/Console/IndexCommand.php]: Please add static type-hints

File: src/Console/IndexCommand.php [hunk 2]
```diff
@@ -123,4 +132,39 @@ protected function getCount()
...
     }
+
+    /**
+     * @param array $responses
+     * @return array
+     */
+    protected function getErrors($responses) 
+    {
+        $errors = [];
+        foreach($responses as $type => $response) {
```
Comment (Jalle19) [inline src/Console/IndexCommand.php]: `$type` doesn't seem to be used here, it gets overwritten in the inner loop

**Comment by Jalle19** [comment]
Also at some point please check the style fixes from Style CI. Normally it creates a PR to your branch but it seems that integration doesn't work in this case. If it's too cumbersome I can do it once this is merged.

File: src/Console/IndexCommand.php (hunk 1 above)
Comment (wojo1206) [reply src/Console/IndexCommand.php]: Any particular generator in mind? `Nord\Lumen\Elasticsearch\Search\Aggregation\AggregationCollection` seems reusable.

Comment (Jalle19) [reply src/Console/IndexCommand.php]: I would just make a new class

**Comment by Jalle19** [comment]
Still a couple of style CI and static analysis issues left, can you fix those?

**Comment by Jalle19** [comment]
Awesome, thanks! I'll merge, test and tag on Monday.

**Pull request merged**